### PR TITLE
[LLVMGPU] Switch LLVMGPUVectorDistribute to use iree_gpu.lowering_config

### PR DESCRIPTION
**Pull request by Groverkss**
Most of the changes in the patch are tests. Actual changes:

- LLVMGPUVectorDistribution pipeline now uses iree_gpu.lowering_config
- GPUApplyTilingLevel does not fuse tensor.pad for reduction tiling level

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 1]
```diff
@@ -115,6 +115,17 @@ bool isROCmBackend(IREE::GPU::TargetAttr target) {
   return target.getArch().starts_with("gfx");
 }
 
+static bool needsLoweringConfigPropagation(
+    IREE::Codegen::DispatchLoweringPassPipeline pipeline) {
+  using Pipeline = IREE::Codegen::DispatchLoweringPassPipeline;
+  SmallVector<Pipeline> supportedPipelines = {
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]:
```suggestion
  Pipeline supportedPipelines[] = {
```

File: compiler/src/iree/compiler/Codegen/Common/GPU/GPUApplyTilingLevel.cpp [hunk 2]
```diff
@@ -135,6 +135,12 @@ applyTileAndFuseToEachRoot(RewriterBase &rewriter,
             bool isDestinationOperand)
         -> std::optional<scf::SCFTileAndFuseOptions::ControlFnResult> {
       Operation *owner = originalProducer.getOwner();
+      if (tilingLevel == IREE::GPU::TilingLevel::Reduction &&
```
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/GPU/GPUApplyTilingLevel.cpp]: [optional] While you're here, I think we also want to turn it off for Subgroup level

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 3]
```diff
@@ -115,6 +115,17 @@ bool isROCmBackend(IREE::GPU::TargetAttr target) {
...
+      Pipeline::LLVMGPUTileAndFuse, Pipeline::LLVMGPUVectorDistribute,
+      Pipeline::LLVMGPUPadAndVectorDistribute};
+  return !llvm::any_of(supportedPipelines, [&pipeline](Pipeline choice) {
+    return choice == pipeline;
+  });
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: `llvm::is_contained`?

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 4]
```diff
@@ -351,51 +363,58 @@ setConvolutionVectorDistributionConfig(IREE::GPU::TargetAttr target,
...
   for (int64_t filterDim : convolutionDims->filterLoop) {
-    workgroupTileSizes[filterDim] = 1;
+    reductionTileSizes[filterDim] = 1;
   }
 
-  TileSizesListType tileSizes;
-  tileSizes.push_back(workgroupTileSizes);
+  MLIRContext *context = op.getContext();
+  Builder b(context);
+  SmallVector<NamedAttribute, 1> attrs;
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: You insert 2 elements just after this
```suggestion
  SmallVector<NamedAttribute, 2> attrs;
```

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 5]
```diff
@@ -596,41 +616,50 @@ setMatmulVectorDistributionConfig(IREE::GPU::TargetAttr target,
...
   LLVM_DEBUG(debugPrintContractionInfo("Workgroup tile sizes", op.getNumLoops(),
                                        *contractionDims, workgroupTileSizes));
+  LLVM_DEBUG(debugPrintContractionInfo("Reduction tile sizes", op.getNumLoops(),
+                                       *contractionDims, reductionTileSizes));
 
-  TileSizesListType tileSizes;
-  tileSizes.push_back(workgroupTileSizes);
+  MLIRContext *context = op.getContext();
+  Builder b(context);
+  SmallVector<NamedAttribute, 1> attrs;
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: same here

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp [hunk 6]
```diff
@@ -789,29 +817,36 @@ setAttentionVectorDistributionConfig(IREE::GPU::TargetAttr target,
...
       schedule->nWarpCount * schedule->nTileCount * schedule->nSize;
 
-  // Follow the LLVMGPU convention of keeping all of the tile sizes in one list.
-  workgroupTileSizes[k2Dim] = schedule->kTileCount * schedule->kSize;
+  reductionTileSizes[k2Dim] = schedule->kTileCount * schedule->kSize;
 
-  TileSizesListType tileSizes;
-  tileSizes.push_back(workgroupTileSizes);
+  MLIRContext *context = op.getContext();
+  SmallVector<NamedAttribute, 1> attrs;
```
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: same here

File: compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp (hunk 1 above)
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp]: nit: Add a short comment that these are the pipelines that do not need propagation (at first glance it looks like the opposite based on the helper name)

**Review by kuhar — COMMENTED** [review]
When you land this, could you please add an issue to update the tuner?

**Review by qedawkins — APPROVED** [review]
Nice, LGTM

**Review by kuhar — APPROVED** [review]

**Pull request merged**